### ISSUE TITLE
Proc print output appears messy within display

### ISSUE THREAD

**Issue by Cazziii** [sassoftware/saspy#97]
On creating a print of a data set using the head function, the data is displayed in a justified way and all variable names overlap.
![image](https://user-images.githubusercontent.com/31005308/36438796-d6457f7a-1662-11e8-92fb-b79b4db73b69.png)

**Comment by tomweber-sas** [comment]
That appears to be output of a dataframe, right? Not ODS output? I wonder if it's jupyter display that does that? If you have a dataframe with the same column names and data and aren't even using saspy, does it do the same?
When we return a dataframe instead of ODS output, I don't think there's anything that we could be doing to cause this. BUT, I can't be sure without investigating some. Does this happen to be public data, such that I could import it and try this out with apples to apples? If not, maybe just send me a SASData.contents() output and I can fabricate a table to see what I see.

Thanks!
Tom

**Comment by Cazziii** [comment]
Hi Tom,  This is actually the output of a SAS dataset using the head function, I appear to have missed off the call.  It appears to happen with any wide set of data:
![image](https://user-images.githubusercontent.com/31005308/36440007-826fef62-1666-11e8-8094-688ef2bebbe4.png)

**Comment by Cazziii** [comment]
Outside of SASPy, the data frame has a scroll bar to move along the data.

**Comment by tomweber-sas** [comment]
Ok, interesting. saspy has a 'results=' option where you can specify output (like from the head() method) to be a dataframe, or be SAS's ODS HTML output, or even text output. The default for that is dataframe ('pandas' I think it the actual value for the option), so crimesas.head() in this case is returning a pandas dataframe, which is then being displayed. 
So if you create your own DF with this data and do a head(), it has scroll bars? Then I think I will need to look into this. Maybe there is something I can do maybe. I'll have to dig into that and see what's different.

Out of curiosity, if you add the results= to your sasdata() and do the head(), does it look reasonable?
crimessas = sas.sasdata('tablename', 'libref', **results='HTML'**)    # whatever table/libref 

Thanks,
Tom

**Comment by tomweber-sas** [comment]
Well, I've created a data set with many long named columns (32 char is the limit). And I see scroll bars for both ODS and HTML. But, looking at your output, it seems to be ODS output, not a data frame.
Can you submit your SASSession object, and send the output from that?
I had an issue with someone getting something like this in sas_kernel (no pandas, everything is ods results); it uses saspy under the covers. I wonder if there's something set in your sas session that's affecting the output. something in a config or autoexec (or sa sdefaults for whatever version) that is affecting the ods output.

Hmmm???

**Comment by Cazziii** [comment]
When I submit the SAS Session object I get this:
![image](https://user-images.githubusercontent.com/31005308/36474154-8710647e-16ee-11e8-84cb-1fe3e3fea743.png)

Using the code you suggected above: crimessas = sas.sasdata('tablename', 'libref', results='HTML')  It gives the exact same results.

**Comment by tomweber-sas** [comment]
Yes, that verifies that you are getting the ODS results, not the dataframe. The output above is from the SASData object, not the SASSession object. If you submit the session object I'll get info about versions of saspy and SAS, what access method you're using, ...

I'm thinking there some options set somewhere that are affecting the HTML5 output and rendering, but I haven't been able to find anything in our doc yet, nor been able to reproduce this yet; I get a scroll bar, and all of the column names are spaced out as would be expected.
Can you also submit
```
crimesas.contents()
```
so I can see the structure of the data. Maybe it's not only the column names length, but the width of the columns that factors into this too. With that output, I can simulate the structure of your data set to see if that changes things for me.

Thanks,
Tom

**Comment by Cazziii** [comment]
The output of the contents is as follows:
![image](https://user-images.githubusercontent.com/31005308/36492710-ab91b604-1725-11e8-89ad-e96c134c7886.png)
![image](https://user-images.githubusercontent.com/31005308/36492743-c117cd74-1725-11e8-9d97-586fd11ae8a1.png)

![image](https://user-images.githubusercontent.com/31005308/36492780-d7814194-1725-11e8-8a27-7c67f26cf9dc.png)

**Comment by Cazziii** [comment]
Output from sas session object is as follows:
![image](https://user-images.githubusercontent.com/31005308/36492883-258fb56e-1726-11e8-85c2-c85d34e8f8bb.png)

**Comment by tomweber-sas** [comment]
Awesome, thanks. Looks like you're running current code. I'll see if I can reproduce this given the schema of the table. 
Tom

**Comment by Cazziii** [comment]
Hi Tom,  do you have any updates for this?

**Comment by tomweber-sas** [comment]
I'm sorry, I have not had a chance to dig into this since the last post. I'm trying to switch back to saspy today, and hopefully for the rest of the week. I will mock up some data and see if I can emulate what you're seeing. If not, I can check with the ODS team to see if they know of anything that could cause this. 

There is one other thing you can try which may help me. We can capture the actual HTML5 being returned (and then rendered) with the following. If you write it to a file you can open it in a browser and see if it renders right. Also send it to me and I can compare against the HTML I get which renders fine for me. We can see if there's something different between our html, which could point to something configured different in ODS. Or if they are the same, maybe something about your browser or the code rendering the document.


You can get the actual html and write it to a file (name it whatever.html) and open it and see how it looks. To take the Jupyter renderer out of the picture. 

in python you would do:

import saspy
sas = saspy.SASsession(results='HTML')
log_lst = sas.submit("Proc print data=mydata._df(obs=5); run;")

Then you can do
print(log_lst["LST"]) 
and that will be a complete html document you can cut-n-paste and write to a file and see how the browser will display it.

Here’s a bit of that to see what to expect (sorry, I copied this from another thread, where they were running proc sql - same difference with your proc print above though):
```
log_lst = sas.submit("Proc SQL inobs=20;    Select * from sashelp.cars; Quit;")
print(log_lst["LST"])

<!DOCTYPE html>
<html lang="en" xml:lang="en" xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta charset="utf-8"/>
<meta content="SAS 9.4" name="generator"/>
<title>SAS Output</title>
<style>
/*<![CDATA[*/
.body.c > table, .body.c > pre, .body.c div > table,
.body.c div > pre, .body.c > table, .body.c > pre,
.body.j > table, .body.j > pre, .body.j div > table,
.body.j div > pre, .body.j > table, .body.j > pre,
.body.c p.note, .body.c p.warning, .body.c p.error, .body.c p.fatal,
.body.j p.note, .body.j p.warning, .body.j p.error, .body.j p.fatal,
.body.c > table.layoutcontainer, .body.j > table.layoutcontainer { margin-left: auto; margin-right: auto }
.layoutregion.l table, .layoutregion.l pre, .layoutregion.l p.note,
[… ten tons of more stuff…]
<td class="r data">340</td>
<td class="r data">14</td>
<td class="r data">20</td>
<td class="r data">3825</td>
<td class="r data">104</td>
<td class="r data">179</td>
</tr>
</tbody>
</table>
</div>
</div>
</body>
</html>
```

**Comment by tomweber-sas** [comment]
I mocked up your data and don't see any issue. I get a scroll bar in jupyter and none of the columns overlap. Attaching an html export of the notebook, which github doesn't support; so remove the extra '.txt' so the file is names .html.
[Issue97.html.txt](https://github.com/sassoftware/saspy/files/1768784/Issue97.html.txt)

If you can send me the html document you get (the instructions above), I'll see if there's anything different then I get. You can just email that directly to me if you like.

Thanks,
Tom

**Comment by Cazziii** [comment]
Hi Tom,  This is the code that is being generated from the proc print:
<!DOCTYPE html>
<html lang="en" xml:lang="en" xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta charset="utf-8"/>
<meta content="SAS 9.4" name="generator"/>
<title>SAS Output</title>
<style>
/*<![CDATA[*/
.body.c > table, .body.c > pre, .body.c div > table,
.body.c div > pre, .body.c > table, .body.c > pre,
.body.j > table, .body.j > pre, .body.j div > table,
.body.j div > pre, .body.j > table, .body.j > pre,
.body.c p.note, .body.c p.warning, .body.c p.error, .body.c p.fatal,
.body.j p.note, .body.j p.warning, .body.j p.error, .body.j p.fatal,
.body.c > table.layoutcontainer, .body.j > table.layoutcontainer { margin-left: auto; margin-right: auto }
.layoutregion.l table, .layoutregion.l pre, .layoutregion.l p.note,
.layoutregion.l p.warning, .layoutregion.l p.error, .layoutregion.l p.fatal { margin-left: 0 }
.layoutregion.c table, .layoutregion.c pre, .layoutregion.c p.note,
.layoutregion.c p.warning, .layoutregion.c p.error, .layoutregion.c p.fatal { margin-left: auto; margin-right: auto }
.layoutregion.r table, .layoutregion.r pre, .layoutregion.r p.note,
.layoutregion.r table, .layoutregion.r pre, .layoutregion.r p.note,
.layoutregion.r p.warning, .layoutregion.r p.error, .layoutregion.r p.fatal { margin-right: 0 }
article, aside, details, figcaption, figure, footer, header, hgroup, nav, section { display: block }
html{ font-size: 100% }
.body { margin: 1em; font-size: 13px; line-height: 1.231 }
sup { position: relative; vertical-align: baseline; bottom: 0.25em; font-size: 0.8em }
sub { position: relative; vertical-align: baseline; top: 0.25em; font-size: 0.8em }
ul, ol { margin: 1em 0; padding: 0 0 0 40px }
dd { margin: 0 0 0 40px }
nav ul, nav ol { list-style: none; list-style-image: none; margin: 0; padding: 0 }
img { border: 0; vertical-align: middle }
svg:not(:root) { overflow: hidden }
figure { margin: 0 }
table { border-collapse: collapse; border-spacing: 0 }
.layoutcontainer { border-collapse: separate; border-spacing: 0 }
p { margin-top: 0; text-align: left }
h1.heading1 { text-align: left }
h2.heading2 { text-align: left }
h3.heading3 { text-align: left }
h4.heading4 { text-align: left }
h5.heading5 { text-align: left }
h6.heading6 { text-align: left }
span { text-align: left }
table { margin-bottom: 1em }
td, th { text-align: left; padding: 3px 6px; vertical-align: top }
td[class$="fixed"], th[class$="fixed"] { white-space: pre }
section, article { padding-top: 1px; padding-bottom: 8px }
hr.pagebreak { height: 0px; border: 0; border-bottom: 1px solid #c0c0c0; margin: 1em 0 }
.stacked-value { text-align: left; display: block }
.stacked-cell > .stacked-value, td.data > td.data, th.data > td.data, th.data > th.data, td.data > th.data, th.header > th.header { border: 0 }
.stacked-cell > div.data { border-width: 0 }
.systitleandfootercontainer { white-space: nowrap; margin-bottom: 1em }
.systitleandfootercontainer > p { margin: 0 }
.systitleandfootercontainer > p > span { display: inline-block; width: 100%; white-space: normal }
.batch { display: table }
.toc { display: none }
.proc_note_group, .proc_title_group { margin-bottom: 1em }
p.proctitle { margin: 0 }
p.note, p.warning, p.error, p.fatal { display: table }
.notebanner, .warnbanner, .errorbanner, .fatalbanner,
.notecontent, .warncontent, .errorcontent, .fatalcontent { display: table-cell; padding: 0.5em }
.notebanner, .warnbanner, .errorbanner, .fatalbanner { padding-right: 0 }
.body > div > ol li { text-align: left }
.c { text-align: center }
.r { text-align: right }
.l { text-align: left }
.j { text-align: justify }
.d { text-align: right }
.b { vertical-align: bottom }
.m { vertical-align: middle }
.t { vertical-align: top }
.aftercaption {
    background-color: #fafbfe;
    border-spacing: 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
    padding-top: 4pt;
}
.batch > colgroup {
    border-left: 1px solid #c1c1c1;
    border-right: 1px solid #c1c1c1;
}
.batch > tbody, .batch > thead, .batch > tfoot {
    border-top: 1px solid #c1c1c1;
    border-bottom: 1px solid #c1c1c1;
}
.batch { border: hidden; }
.batch {
    background-color: #fafbfe;
    border: 1px solid #c1c1c1;
    border-collapse: separate;
    border-spacing: 1px;
    color: #000000;
    font-family: 'SAS Monospace', 'Courier New', Courier, monospace;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    padding: 7px;
    }
.beforecaption {
    background-color: #fafbfe;
    border-spacing: 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.body {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    margin-left: 8px;
    margin-right: 8px;
}
.bodydate {
    background-color: #fafbfe;
    border-spacing: 0;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    text-align: right;
    vertical-align: top;
    width: 100%;
}
.bycontentfolder {
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: none;
    margin-left: 6pt;
}
.byline {
    background-color: #fafbfe;
    border-spacing: 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.bylinecontainer > col, .bylinecontainer > colgroup > col, .bylinecontainer > colgroup, .bylinecontainer > tr, .bylinecontainer > * > tr, .bylinecontainer > thead, .bylinecontainer > tbody, .bylinecontainer > tfoot { border: none; }
.bylinecontainer {
    background-color: #fafbfe;
    border: none;
    border-spacing: 1px;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    width: 100%;
}
.caption {
    background-color: #fafbfe;
    border-spacing: 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.cell, .container {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.contentfolder, .contentitem {
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: none;
    margin-left: 6pt;
}
.contentproclabel, .contentprocname {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.contents {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: decimal;
    margin-left: 8px;
    margin-right: 8px;
}
.contentsdate {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    width: 100%;
}
.contenttitle {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: italic;
    font-weight: bold;
}
.continued {
    background-color: #fafbfe;
    border-spacing: 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
    width: 100%;
}
.data, .dataemphasis {
    background-color: #ffffff;
    border-color: #c1c1c1;
    border-style: solid;
    border-width: 0 1px 1px 0;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.dataemphasisfixed {
    background-color: #ffffff;
    border-color: #c1c1c1;
    border-style: solid;
    border-width: 0 1px 1px 0;
    font-family: 'Courier New', Courier, monospace;
    font-size:  normal;
    font-style: italic;
    font-weight: normal;
}
.dataempty {
    background-color: #ffffff;
    border-color: #c1c1c1;
    border-style: solid;
    border-width: 0 1px 1px 0;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.datafixed {
    background-color: #ffffff;
    border-color: #c1c1c1;
    border-style: solid;
    border-width: 0 1px 1px 0;
    font-family: 'Courier New', Courier;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.datastrong {
    background-color: #ffffff;
    border-color: #c1c1c1;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.datastrongfixed {
    background-color: #ffffff;
    border-color: #c1c1c1;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #000000;
    font-family: 'Courier New', Courier, monospace;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.date {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    width: 100%;
}
.document {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.errorbanner {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.errorcontent {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.errorcontentfixed {
    background-color: #fafbfe;
    color: #112277;
    font-family: 'Courier New', Courier;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.extendedpage {
    background-color: #fafbfe;
    border-style: solid;
    border-width: 1pt;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: italic;
    font-weight: normal;
    text-align: center;
}
.fatalbanner {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.fatalcontent {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.fatalcontentfixed {
    background-color: #fafbfe;
    color: #112277;
    font-family: 'Courier New', Courier;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.folderaction {
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: none;
    margin-left: 6pt;
}
.footer {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.footeremphasis {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: italic;
    font-weight: normal;
}
.footeremphasisfixed {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: 'Courier New', Courier, monospace;
    font-size:  normal;
    font-style: italic;
    font-weight: normal;
}
.footerempty {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.footerfixed {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: 'Courier New', Courier;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.footerstrong {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.footerstrongfixed {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: 'Courier New', Courier, monospace;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.frame {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.graph > colgroup {
    border-left: 1px solid #c1c1c1;
    border-right: 1px solid #c1c1c1;
}
.graph > tbody, .graph > thead, .graph > tfoot {
    border-top: 1px solid #c1c1c1;
    border-bottom: 1px solid #c1c1c1;
}
.graph { border: hidden; }
.graph {
    background-color: #fafbfe;
    border: 1px solid #c1c1c1;
    border-collapse: separate;
    border-spacing: 1px;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    }
.header {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.headeremphasis {
    background-color: #d8dbd3;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: italic;
    font-weight: normal;
}
.headeremphasisfixed {
    background-color: #d8dbd3;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #000000;
    font-family: 'Courier New', Courier, monospace;
    font-size:  normal;
    font-style: italic;
    font-weight: normal;
}
.headerempty {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.headerfixed {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: 'Courier New', Courier;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.headersandfooters {
    background-color: #edf2f9;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.headerstrong {
    background-color: #d8dbd3;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.headerstrongfixed {
    background-color: #d8dbd3;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #000000;
    font-family: 'Courier New', Courier, monospace;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.heading1, .heading2, .heading3, .heading4, .heading5, .heading6 { font-family: Arial, Helvetica, sans-serif }
.index {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.indexaction, .indexitem {
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: none;
    margin-left: 6pt;
}
.indexprocname {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.indextitle {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: italic;
    font-weight: bold;
}
.layoutcontainer, .layoutregion {
    border-width: 0;
    border-spacing: 30px;
}
.linecontent {
    background-color: #fafbfe;
    border-color: #c1c1c1;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.list {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: disc;
}
.list10 {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: square;
}
.list2 {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: circle;
}
.list3, .list4, .list5, .list6, .list7, .list8, .list9 {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: square;
}
.listitem {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: disc;
}
.listitem10 {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: square;
}
.listitem2 {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: circle;
}
.listitem3, .listitem4, .listitem5, .listitem6, .listitem7, .listitem8, .listitem9 {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: square;
}
.note {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.notebanner {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.notecontent {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.notecontentfixed {
    background-color: #fafbfe;
    color: #112277;
    font-family: 'Courier New', Courier;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.output > colgroup {
    border-left: 1px solid #c1c1c1;
    border-right: 1px solid #c1c1c1;
}
.output > tbody, .output > thead, .output > tfoot {
    border-top: 1px solid #c1c1c1;
    border-bottom: 1px solid #c1c1c1;
}
.output { border: hidden; }
.output {
    background-color: #fafbfe;
    border: 1px solid #c1c1c1;
    border-collapse: separate;
    border-spacing: 1px;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    }
.pageno {
    background-color: #fafbfe;
    border-spacing: 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
    text-align: right;
    vertical-align: top;
}
.pages {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: decimal;
    margin-left: 8px;
    margin-right: 8px;
}
.pagesdate {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    width: 100%;
}
.pagesitem {
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    list-style-type: none;
    margin-left: 6pt;
}
.pagesproclabel, .pagesprocname {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.pagestitle {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: italic;
    font-weight: bold;
}
.paragraph {
    background-color: #fafbfe;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.parskip > col, .parskip > colgroup > col, .parskip > colgroup, .parskip > tr, .parskip > * > tr, .parskip > thead, .parskip > tbody, .parskip > tfoot { border: none; }
.parskip {
    border: none;
    border-spacing: 0;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
    }
.prepage {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    text-align: left;
}
.proctitle {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.proctitlefixed {
    background-color: #fafbfe;
    color: #112277;
    font-family: 'Courier New', Courier, monospace;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.rowfooter {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.rowfooteremphasis {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: italic;
    font-weight: normal;
}
.rowfooteremphasisfixed {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: 'Courier New', Courier, monospace;
    font-size:  normal;
    font-style: italic;
    font-weight: normal;
}
.rowfooterempty {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.rowfooterfixed {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: 'Courier New', Courier;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.rowfooterstrong {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.rowfooterstrongfixed {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: 'Courier New', Courier, monospace;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.rowheader {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.rowheaderemphasis {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: italic;
    font-weight: normal;
}
.rowheaderemphasisfixed {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: 'Courier New', Courier, monospace;
    font-size:  normal;
    font-style: italic;
    font-weight: normal;
}
.rowheaderempty {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.rowheaderfixed {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: 'Courier New', Courier;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.rowheaderstrong {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.rowheaderstrongfixed {
    background-color: #edf2f9;
    border-color: #b0b7bb;
    border-style: solid;
    border-width: 0 1px 1px 0;
    color: #112277;
    font-family: 'Courier New', Courier, monospace;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.systemfooter, .systemfooter10, .systemfooter2, .systemfooter3, .systemfooter4, .systemfooter5, .systemfooter6, .systemfooter7, .systemfooter8, .systemfooter9 {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.systemtitle, .systemtitle10, .systemtitle2, .systemtitle3, .systemtitle4, .systemtitle5, .systemtitle6, .systemtitle7, .systemtitle8, .systemtitle9 {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size: small;
    font-style: normal;
    font-weight: bold;
}
.systitleandfootercontainer > col, .systitleandfootercontainer > colgroup > col, .systitleandfootercontainer > colgroup, .systitleandfootercontainer > tr, .systitleandfootercontainer > * > tr, .systitleandfootercontainer > thead, .systitleandfootercontainer > tbody, .systitleandfootercontainer > tfoot { border: none; }
.systitleandfootercontainer {
    background-color: #fafbfe;
    border: none;
    border-spacing: 1px;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    width: 100%;
}
.table > col, .table > colgroup > col {
    border-left: 1px solid #c1c1c1;
    border-right: 0 solid #c1c1c1;
}
.table > tr, .table > * > tr {
    border-top: 1px solid #c1c1c1;
    border-bottom: 0 solid #c1c1c1;
}
.table { border: hidden; }
.table {
    border-color: #c1c1c1;
    border-style: solid;
    border-width: 1px 0 0 1px;
    border-collapse: collapse;
    border-spacing: 0;
    }
.titleandnotecontainer > col, .titleandnotecontainer > colgroup > col, .titleandnotecontainer > colgroup, .titleandnotecontainer > tr, .titleandnotecontainer > * > tr, .titleandnotecontainer > thead, .titleandnotecontainer > tbody, .titleandnotecontainer > tfoot { border: none; }
.titleandnotecontainer {
    background-color: #fafbfe;
    border: none;
    border-spacing: 1px;
    color: #000000;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
    width: 100%;
}
.titlesandfooters {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.usertext {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.warnbanner {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: bold;
}
.warncontent {
    background-color: #fafbfe;
    color: #112277;
    font-family: Arial, 'Albany AMT', Helvetica, Helv;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
.warncontentfixed {
    background-color: #fafbfe;
    color: #112277;
    font-family: 'Courier New', Courier;
    font-size:  normal;
    font-style: normal;
    font-weight: normal;
}
/*]]>*/
</style>
</head>
<body class="l body">
<div style="padding-bottom: 8px; padding-top: 1px">
<div id="IDX" class="systitleandfootercontainer" style="border-spacing: 1px">
<p><span class="c systemtitle">The SAS System</span> </p>
</div>
<div style="padding-bottom: 8px; padding-top: 1px">
<table class="table" style="border-spacing: 0" aria-label="Data Set MYDATA._DF">
<caption aria-label="Data Set MYDATA._DF"></caption>
<colgroup><col/></colgroup><colgroup><col/><col/><col/><col/><col/><col/><col/><col/><col/><col/><col/><col/></colgroup>
<thead>
<tr>
<th class="r header" scope="col">Obs</th>
<th class="header" scope="col">age_range</th>
<th class="r header" scope="col">datetime</th>
<th class="header" scope="col">gender</th>
<th class="header" scope="col">involved_person</th>
<th class="header" scope="col">legislation</th>
<th class="header" scope="col">object_of_search</th>
<th class="header" scope="col">officer_defined_ethnicity</th>
<th class="header" scope="col">outcome</th>
<th class="header" scope="col">outcome_link_to_object</th>
<th class="header" scope="col">removed_clothing</th>
<th class="r header" scope="col">latitude</th>
<th class="r header" scope="col">longitude</th>
</tr>
</thead>
<tbody>
<tr>
<th class="r rowheader" scope="row">1</th>
<td class="data">25-34</td>
<td class="r data">2017-07-01T00:00:00.000000</td>
<td class="data">Male</td>
<td class="data">True</td>
<td class="data">Misuse of Drugs Act 1971 (section 23)</td>
<td class="data">Controlled drugs</td>
<td class="data">White</td>
<td class="data">False</td>
<td class="data">nan</td>
<td class="data">False</td>
<td class="r data">51.7397</td>
<td class="r data" style="white-space: nowrap">-0.96730</td>
</tr>
<tr>
<th class="r rowheader" scope="row">2</th>
<td class="data">25-34</td>
<td class="r data">2017-07-01T00:00:00.000000</td>
<td class="data">Male</td>
<td class="data">True</td>
<td class="data">Misuse of Drugs Act 1971 (section 23)</td>
<td class="data">Controlled drugs</td>
<td class="data">White</td>
<td class="data">False</td>
<td class="data">nan</td>
<td class="data">False</td>
<td class="r data">51.7397</td>
<td class="r data" style="white-space: nowrap">-0.96730</td>
</tr>
<tr>
<th class="r rowheader" scope="row">3</th>
<td class="data">18-24</td>
<td class="r data">2017-07-01T00:00:00.000000</td>
<td class="data">Male</td>
<td class="data">True</td>
<td class="data">Misuse of Drugs Act 1971 (section 23)</td>
<td class="data">Controlled drugs</td>
<td class="data">White</td>
<td class="data">False</td>
<td class="data">nan</td>
<td class="data">False</td>
<td class="r data">51.7397</td>
<td class="r data" style="white-space: nowrap">-0.96730</td>
</tr>
<tr>
<th class="r rowheader" scope="row">4</th>
<td class="data">10-17</td>
<td class="r data">2017-07-01T04:03:00.000000</td>
<td class="data">Male</td>
<td class="data">True</td>
<td class="data">Misuse of Drugs Act 1971 (section 23)</td>
<td class="data">Controlled drugs</td>
<td class="data">Black</td>
<td class="data">False</td>
<td class="data">nan</td>
<td class="data">False</td>
<td class="r data">51.5078</td>
<td class="r data" style="white-space: nowrap">-0.59830</td>
</tr>
<tr>
<th class="r rowheader" scope="row">5</th>
<td class="data">10-17</td>
<td class="r data">2017-07-01T04:03:00.000000</td>
<td class="data">Male</td>
<td class="data">True</td>
<td class="data">Misuse of Drugs Act 1971 (section 23)</td>
<td class="data">Controlled drugs</td>
<td class="data">Black</td>
<td class="data">False</td>
<td class="data">nan</td>
<td class="data">False</td>
<td class="r data">51.5078</td>
<td class="r data" style="white-space: nowrap">-0.59830</td>
</tr>
</tbody>
</table>
</div>
</div>
</body>
</html>

**Comment by tomweber-sas** [comment]
Well, lets try this again. Pasting it into this interface didn't get what I need. Though it did render the table without any overlap. Can you submit the following (change the file system path to whatever you like), and email me the file itself? Also, when you open that file (should open in a browser given the .html extension), does it render right?

ll = sas.submit('''proc print data=mydata._df(obs=5);run;''')
fd = open('C:\\New folder\\procprint.html', 'w')
fd.write(ll['LST'])
fd.close()

Thanks,
Tom

**Comment by Cazziii** [comment]
This appears to render sucessfully:
![image](https://user-images.githubusercontent.com/31005308/36911879-d7ce748c-1e3c-11e8-827a-9d092c5d6f90.png)

I will send the html file separately via email.

**Comment by tomweber-sas** [comment]
Thanks, yes I got the file and it's no different then what I'm seeing when I run this too. I assume you're using the same browser to open the file as you're running Jupyter in? Are you running this in Jupyter in a browser? I don't see this above, guess I'm assuming so. Or are you using some other UI interface for python?
I'm at a loss to identify the problem. I have to think it's an issue with the application rendering the html.
The html seems fine and I don't see any directives in it that would make columns fixed with, such that they overlap. I get the scroll bar in Juptyer and in the browser directly (if the window isn't wide enough to show it all).
???,
Tom

**Comment by Cazziii** [comment]
The code was run and then opened in the same browser. Strangely this time is rendered correctly.  However, within Jupyter it does not.  I am using Google Chrome browser for both.

**Comment by Cazziii** [comment]
IPython version is as follows (Appears much higher than the version that you have):
![image](https://user-images.githubusercontent.com/31005308/37022002-76bfa834-2119-11e8-8c05-9bec9a8fa3b5.png)

**Comment by tomweber-sas** [comment]
Yep, that's a newer version. So, to try it out I just did an update and it installed 6.2.1. Unfortunately, jupyter's now broke. I'm going to have to get this back working again. This is probably a python3.6 module and I have 3.5 installed. Probably gonna have to clean it all out and reinstall. Guess I could move up to 3.6.

**Comment by tomweber-sas** [comment]
Well, I just got the latest anaconda installed:
```
c:\>pip show ipython
Name: ipython
Version: 6.2.1
Summary: IPython: Productive Interactive Computing
Home-page: https://ipython.org
Author: The IPython Development Team
Author-email: ipython-dev@python.org
License: BSD
Location: c:\programdata\anaconda3\lib\site-packages
Requires: setuptools, jedi, decorator, pickleshare, simplegeneric, traitlets, prompt-toolkit, pygments, colorama

c:\>python --version
Python 3.6.4 :: Anaconda, Inc.

c:\>
```
And now I get the same broken behavior you're seeing. Well, at least I'm now seeing what you're seeing.
This all worked with the version I had before, so I'll try to see if I can figure out what happened.

It seems pretty clear to me this is with ipython or some part of jupyter that's changed since the previous version. Nothing's different with saspy or SAS or the HTML that's being rendered. 

Guess this is progress :)   I liked the previous version better :(
Tom

**Comment by tomweber-sas** [comment]
Well, I've played around with this a bit, trying the various display methods and can't get it to render correctly. So, I've entered an issue with ipython (https://github.com/ipython/ipython/issues/11033). Hopefully I'll hear something back and get a fix from them.

**Comment by tomweber-sas** [comment]
The above issue I opened had me go to the notebook part of Ipython, to an issue that seemed to be the cause of this. But that issue has been closed for a year. I posted to it but haven't seen a response. So I opened a new issue, https://github.com/jupyter/notebook/issues/3413 a week ago, but have yet to get a response. This is actually my first time opening an issue with an open source project. I'm not really sure what to expect. Crossing my fingers.
Tom

**Comment by tomweber-sas** [comment]
I've still had no response from the issue I opened with jupyter notebooks (3431). But, I have had confirmation from another user who experienced this same problem a month or two ago, that upgrading to JupyterLab, which is now 'production' as opposed to Beta, did resolve this. It' not broken in that release. 

Jared's been using Lab for a while with no problems, and I'd still been using an earlier version of Notebook with no problem, until upgrading to this version investigating this issue. It appears that the only resolution to this will end up being upgrading or downgrading from this version where table rendering is broken. The only other thing, for saspy at least, but not sas_kernel, is to have table returned as pandas data frames. Data Frames render, for me, in this version even though html tables do not.

@Cazziii are you able to try a different version to see that this solves this problem for you?

Tom

**Comment by Cazziii** [comment]
Hi Tom,
I have downgraded my version to 5.3.0 and am still finding the same issue:
![image](https://user-images.githubusercontent.com/31005308/37660762-0f4cb240-2c4b-11e8-8011-e5d5dca20bbc.png)
The page still looks as follows:
![image](https://user-images.githubusercontent.com/31005308/37660858-4e8e42f2-2c4b-11e8-967f-2c4d5ae102bc.png)
Many Thanks,

**Comment by tomweber-sas** [comment]
Hey @Cazziii did you just downgrade that one ipython module, or reinstall the entire older version of jupyter? I really have no idea where in all of that code this is broke. Do you have an Anaconda install, or just a python install with jupyter installed to? I had to reinstall the older anaconda on my pc, replacing the newer one I installed and saw the problem in. On my linux system, I didn't have anaconda, just individual installs of python, jupyter, ...  and I couldn't revert back to what I had when I upgraded ipython and it broke jupyter (that was a different problem), so I had to upgrade the whole jupyter to get it working again, which means I have the broken version (but it runs). 

It looks like there's a Notebook package too, And from what I did get from the ipython issue; pointed me to notebook, it may be in that piece. But again, I don't know what all pieces parts there are to all of juptyer and where the bug really exists. 

So, If you only did ipython, I am not surprised it didn't solve it. But if you did all of juptyer, or anaconda, then that should get you to a clean release of jupyter.

On my pc, the anaconda I have has Notebook at 4.3.1 (works) while on Linux Notebook is at 5.4.0 (broken). But there are lots of interdependencies, so, I don't know whether you can get away with changing just parts of it or not.

Tom

**Comment by Cazziii** [comment]
Hi Tom,
I downgraded just the IPython module and it made no difference.  I have an anaconda install of Jupyter so I will look at removing anaconda completely and reinstalling anaconda 4.3.1.  I will update you on whether this works.

Thanks,

**Comment by tomweber-sas** [comment]
I expect that to work. Also, and I haven't done this myself personally, but Jared has, the new version of the whole jupyter notbook thing is now jupyter labs (or Lab instead of Notebook). That also works well. Both Jared and another customer have run that successfully. That's the 'newer' upgrade path as opposed to down grade path. Just FYI.

Thanks,
Tom

**Comment by Cazziii** [comment]
Hi Tom,
I have now reinstalled Anaconda3 Version 5.1.0.
This is using IPython version 6.2.1
![image](https://user-images.githubusercontent.com/31005308/37681905-e87f99cc-2c7f-11e8-8e44-ecadf5f92817.png)
I can now see the print working sucessfully and rendering in the format of a data frame:
![image](https://user-images.githubusercontent.com/31005308/37682014-350b12e4-2c80-11e8-8ae5-201114846212.png)
However, I am now experiencing issues with the Contents Procedure not rendering at all (Top of the screen shows as below):
![image](https://user-images.githubusercontent.com/31005308/37682058-600407e4-2c80-11e8-911d-a85f304dc9f0.png)
Describe and bar functions render correctly using SAS so it may be isolated to this procedure.
Many Thanks,

**Comment by tomweber-sas** [comment]
Wait, isn't that the broken version? They didn't seem to break Data Frame rendering, just HTML table rendering. Have you tried displaying a table in HTML?

**Comment by Cazziii** [comment]
Hi Tom,
I have upgraded to JupyterLab, I was not aware that this existed but it appears to be rendering all of the procedure correctly now.  Thank you for your help on this, you can now close the track.  See you at the Global Forum?
Many Thanks!
Carrie

**Comment by tomweber-sas** [comment]
Great, yes Jared has been using Lab for a while now (Beta), and I believe it's 'production' now. He's seen much better rendering with that. I'm glad you were able to switch and this is resolved.
I won't be at SGF this year, but Jared will. In fact he'll be doing a Super Demo of saspy this year. We haven't done much promotion of this before, so this will be the first big thing at SGF for saspy.
Catch up with him there for sure!

Thanks,
Tom

**Comment by jld23** [comment]
@Cazziii, I will be doing a superdemo at SAS Global Forum on Monday at 2 and again Wednesday at 1. I'd love to catch up. If those time don't work, look for me at the programming booth or send me a message.

Jared
